top-crates/src/main.rs

```diff
@@ -53,7 +54,24 @@ fn main() {
     let modifications: Modifications =
         toml::from_slice(&d).expect("unable to parse crate modifications file");
 
-    let (dependencies, infos) = rust_playground_top_crates::generate_info(&modifications);
```

Comment:
This is the only part that I have qualms about. Niko and team were [doing some analysis](https://github.com/optopodi/optopodi) and requested that this code be made available so that they could get (more-or-less) the same list as the playground.

I don't mind changing the public API at all, but it would probably be nice if we kept a simpler API for them.

Reply:
Ah okay, did not know. The only reason I pulled `generate_info` out so that the loop lives in main.rs instead of lib.rs was so that I could put `println!("resolve #{i}")` in there. It felt weird for the println to be in the lib, and main was already printing things using println.

Would it be okay to move the big green block below this (`let config` through `let infos`) back into lib.rs as `generate_info`, even with the println? Or take out the println? Or keep it here, but put another copy without println into lib.rs for Niko and friends? Something else?

Reply:
I just removed b56e90a6ec3eb1315bbe75d606683c67d33f3810 entirely from the PR. This runs fast enough that progress reporting isn't important.